addons/sale_mrp/models/account_move.py

```diff
@@ -11,8 +11,10 @@ def _stock_account_get_anglo_saxon_price_unit(self):
 
         so_line = self.sale_line_ids and self.sale_line_ids[-1] or False
         if so_line:
-            bom = (so_line.product_id.variant_bom_ids or so_line.product_id.product_tmpl_id.bom_ids).filtered(
-                lambda b: not b.company_id or b.company_id == so_line.company_id
+            bom = so_line.product_id.variant_bom_ids.filtered(
+                lambda b: (not b.company_id or b.company_id == so_line.company_id)
+            )[:1] or so_line.product_id.product_tmpl_id.bom_ids.filtered(
+                lambda b: (not b.company_id or b.company_id == so_line.company_id) and not b.product_id
```

Comment:
Hola ! Maybe use `_bom_find_domain` instead :
```suggestion
            kit = self.env['mrp.bom']._bom_find(
                so_line.product_id, company_id=so_line.company_id, bom_type='phantom'
            )[:1]
```
Also, adding a test for the case explained in the commit's message could be usefull :)

Reply:
Thanks for the suggestion.
As for the test, there's already test_12_sale_mrp_anglo_saxon_variant for that.

Reply:
Yeah but if the test passed without your fix, it need to be modify to be sure it will catch the error the next time.
Also, in the `test_12_sale_mrp_anglo_saxon_variant`, both products have a BoM (and none of them is a kit) but in your case, the issue happens only when V1 has no BoM and V2 has a BoM kit.
So it's not the same usecase and I think it should be tested

Reply:
test added